src/components/Traceroute/Traceroute.tsx

```diff
@@ -1,52 +1,40 @@
 import { useForm } from "@mantine/form";
-import { useCallback, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
 import { UserGuide } from "../UserGuide/UserGuide";
-import { SaveOutputToTextFile } from "../SaveOutputToFile/SaveOutputToTextFile";
-
-const title = "Traceroute Tool";
-const description_userguide =
-    "The Traceroute tool provides a utility for displaying the route that IP packets have used as they travel to a " +
-    "particular network or host. For these routes, where possible, the tool will display both the IP address and hostname " +
-    "for any machines visited along the route by the packets. This tool proves useful for network debugging where traceroute " +
-    "will show where the issue is located along the route.\n\nFurther information can be found at: https://www.kali.org/tools" +
-    "/traceroute/\n\n" +
-    "Using Traceroute:\n" +
-    "You will need to upgrade your terminal to a root terminal by typing your password to the terminal you used to open DDT.\n\n" +
-    "If you chose one of these three scans: ICMP, TCP, or UDP, you only need to provide a hostname.\n\n" +
-    "If you chose custom scan, you can type your customized traceroute option.\n" +
-    "For example: -I -p 12345\n\n" +
-    "Step 1: Enter a Hostname/IP address.\n" +
-    "       Eg: 192.168.0.1\n\n" +
-    "Step 2: (Optional) Enter any additional options.\n\n" +
-    "Step 3: Select a scan option.\n" +
-    "       Eg: Traceroute UDP scan\n\n" +
-    "Step 4: Click Scan to commence Traceroute operation.\n\n" +
-    "Step 5: View the Output block below to view the results of the tool's execution.";
-
-//Variables
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import { RenderComponent } from "../UserGuide/UserGuide";
+
+/**
+ * FormValuesType defines the structure for the form values used in the TracerouteTool component.
+ * @field hostname: The hostname or IP address for the traceroute operation.
+ * @field portNumber:The port number to be used, currently not utilized in the traceroute operations.
+ * @field tracerouteSwitch: The selected type of traceroute scan.
+ * @field tracerouteOptions: Custom traceroute options provided by the user.
+ */
 interface FormValuesType {
     hostname: string;
     portNumber: string;
     tracerouteSwitch: string;
     tracerouteOptions: string;
 }
 
-//Traceroute Options
-const tracerouteSwitch = [
-    "Traceroute ICMP scan",
-    "Traceroute TCP scan",
-    "Traceroute UDP scan",
-    "Traceroute custom scan",
-];
-
-//Tool name must be capital or jsx will cry out errors :P
+// Component State Variables.
 const TracerouteTool = () => {
```

Comment:
Add docstring. 
```suggestion
/**
 * The Traceroute component.
 * @returns The Traceroute component.
 */
const TracerouteTool = () => {
```

Reply:
Rename component Traceroute
```suggestion
const Traceroute = () => {
```

Reply:
I have done both of those added the doc string and renamed the tool just Traceroute.